### ISSUE TITLE
LSP reports false error when any field of struct has `#[cfg]` attr.

### ISSUE THREAD

**Issue by blyxyas** [rust-lang/rust#111281]
I don't know if this should be filled on rust-analyzer or here. Feel free to move.
I tried this code:

```rust
#[derive(Clone)] // <- Error here
struct S {
	#[cfg(feature = "a_feature")]
	hi: String
}

fn main() {}
```

I expected to see this happen: *Nothing*

Instead, this happened: Rust-analyzer triggers on `#[Derive(Clone)]` with the message `no such field`. This happens when any field of a struct is behind a `#[cfg]`. Note that using commands like `cargo check` or `cargo run` won't report this error. Only the LSP will report it.

Obviously, if `a_feature` is enabled by default, the error dissapears.

### Meta

`rustc --version --verbose`:
```
rustc 1.71.0-nightly (9ecda8de8 2023-04-30)
binary: rustc
commit-hash: 9ecda8de85ce893cc3fc748ab06be0b8250147a7
commit-date: 2023-04-30
host: x86_64-unknown-linux-gnu
release: 1.71.0-nightly
LLVM version: 16.0.2
```

<details><summary>Backtrace</summary>
<p>

**N/A**

</p>
</details>

**Comment by clubby789** [comment]
I can't reproduce this (and this should probably be moved to the R-A repo). What R-A version are you using?

**Comment by blyxyas** [comment]
`0.4.1503`, pre-release

**Comment by fmease** [comment]
Yes, this should definitely be reported over at https://github.com/rust-lang/rust-analyzer.
I'm closing this as this cannot be reproduced apparently and since we don't have many r-a experts around here. If you still experience this issue I recommend you to search the issue tracker of the aforementioned repo.
